auth/policies.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package auth
+
+import (
+	"context"
+
+	acl "github.com/ory/keto/proto/ory/keto/acl/v1alpha1"
+)
+
+// Authz represents a authorization service. It exposes
+// functionalities through `auth` to perform authorization.
+type Authz interface {
+	// Authorize checks authorization of the given `subject`. Basically,
+	// Authorize verifies that Is `subject` allowed to `relation` on
+	// `object`. Authorize returns the bool indicating the authorization
+	// of the given subject and the error. For example, the response as:
+	// `false, nil` means that `subject` has not `relation` on the `object`.
+	// Therefore, this response means that incoming request is denied.
+	Authorize(ctx context.Context, subject, object, relation string) (bool, error)
+
+	// AddPolicy creates a policy for the given subject. So that, after
+	// AddPolicy, `subject`has a `relation` on `object`. Returns non-nil
+	// error in case of failures.
+	AddPolicy(ctx context.Context, subject, object, relation string) error
+}
+
+// PolicyAgent facilitates the communication to authorization
+// services and implements Authz functionalities for certain
+// authorization services (e.g. ORY Keto).
+type PolicyAgent interface {
+	// CheckPolicy checks if the subject has a relation on the object.
+	// It returns AuthorizationResult consisting of one boolean which indicates
+	// result of the authorization (true: Allowed, false: Denied) and error.
```

Comment:
Similar to @dusanb94 [remark](https://github.com/mainflux/mainflux/pull/1440/files#r686922624) do we need bool here or error that has none nill value is enough (can contain error message also, would be really helpful for logs)?